packages/gatsby-core-utils/src/site-metadata.ts

```diff
@@ -0,0 +1,30 @@
+import { createServiceLock, getService } from "./service-lock"
+
+export interface ISiteMetadata {
+  sitePath: string
+  name?: string
+  pid?: number
+  lastRun?: number
+}
+
+export async function getSiteMetadata(
+  sitePath: string
+): Promise<ISiteMetadata | null> {
+  return getService(sitePath, `metadata`, true)
+}
+
+export async function updateSiteMetadata(
+  metadata: ISiteMetadata,
+  merge = true
```

Comment:
Nit

I would kind of suggest to not use a default and to use constants to prevent magic variables and "the boolean trap"-problem.

```js
export const AND_MERGE = true
export const NOT_MERGE = false
```
or smth, or to export a version of this function that passes on false for you

```js
export async function updateSiteMetadataAndMerge(
  metadata: ISiteMetadata
) {
  return updateSiteMetadata(metadata, true)
}
```
etc